### PR TITLE
feat: Add Issue Supervision Flows

### LINKED ISSUE #446: [Advanced]: Add Workflow to Triage Issues
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently when a contributor creates an issue no one is notified and the issue remains open. 

This creates a problem as the team needs to determine if the issue is valid and label it properly.
With out proper labeling our current bots to do not count the work. This interferes with issue counts for contributors. Ultimately, the bot does not recognize the issue or add it to the count with proper triage.  

### 🛠️ Implementation Notes

We propose to implement work flow files that will moderate issues by reserving/locking them until the issue has been approved by a team member. 

Workflows must be tested and corresponding PR's must include some form of successful testing to ensure code works as intended. 

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by aceppaluni**
# Description

This PR aims to add workflows to assist with triaging issues. The goal is to close off issues from interaction until they have been approved by a supporting team member

## Changes Made
- [x] Added...  moderate-new-issues.yml and approved-issues.yml
- [ ] Modified...
- [ ] Fixed...

## Related Issues

Fixes: #446 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [x] Branch up-to-date with main

## Deployment Notes

A link to testing: https://github.com/aceppaluni/hiero-website/issues/9

## Summary by CodeRabbit

* **New Features**
  * Added automatic moderation for newly opened issues, including a pending review label, a status comment, and issue locking until review.
  * Added an approval flow for issues labeled approved that removes the review label, unlocks the issue, and posts an approval update.

### CODE REVIEW

**Comment by danielmarv** [comment]
@aceppaluni thanks
could you please update the branch

File: .github/workflows/moderate-new-issues.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+permissions:
+  issues: write
+
+concurrency:
+  group: moderate-issues-${{ github.event.issue.number }}
+  cancel-in-progress: false
+
+jobs:
+  moderate:
+    name: Moderate Issue
```
Comment (danielmarv) [inline .github/workflows/moderate-new-issues.yml:23]: Please skip moderation for people who already have access, so we don't lock ourselves out of our own issues:
ie but you can replace with the real teams
```yaml
if: >-
  !contains(fromJSON('["OWNER","MEMBER","COLLABORATOR"]'),
            github.event.issue.author_association)
```

File: .github/workflows/approve-issues.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Harden Runner
+        uses: step-security/harden-runner@fe104658747b27e96e4f7e80cd0a94068e53901d # v2.16.1
+        with:
+          egress-policy: audit
+
+      - name: Remove pending-review label
+        uses: actions-ecosystem/action-remove-labels@2ce5d41b4b6aa8503e285553f75ed56e0a40bae0 # v1.3.0
+        with:
+          labels: pending-review
+
```
Comment (danielmarv) [inline .github/workflows/approve-issues.yml:38]: This action (and `action-remove-labels` in the other file) is set to `using: node12`. Node 12 was removed from Actions runners years ago — these only still run because GitHub silently swaps in Node 20. Neither action has had a release since 2021, so they'll break eventually and nobody will fix them.

You're already using `gh api` in this PR, so we don't need them at all:

```yaml
- name: Flag issue as pending review
  run: gh issue edit "$ISSUE_NUMBER" --repo "$REPO" --add-label pending-review
  env:
    GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    REPO: ${{ github.repository }}
    ISSUE_NUMBER: ${{ github.event.issue.number }}
```

File: .github/workflows/moderate-new-issues.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            Thanks for your submission.
+
+            This issue is awaiting approval from @hiero-ledger/hiero-website-committers and @hiero-ledger/hiero-website-maintainers.
+            
+      - name: Lock issue (IMPORTANT FIX)
+        run: |
+          gh api \
+            -X PUT \
+            "repos/${REPO}/issues/${ISSUE_NUMBER}/lock" \
+            -f lock_reason=off-topic
```
Comment (danielmarv) [inline .github/workflows/moderate-new-issues.yml:58]: `approve-issues.yml` has no `concurrency` block, so the two workflows can run at the same time.

If someone adds `approved` right after the issue opens, approve can finish first — it removes a label that isn't there yet and unlocks an issue that isn't locked yet. Then this workflow locks it. Result: a locked issue with no `pending-review` label, and no automated way to unlock it.

Fix: use the same group name in both files.

```yaml
concurrency:
  group: issue-moderation-${{ github.event.issue.number }}
  cancel-in-progress: false
```

**Review by danielmarv — APPROVED** [review]
